sys/cddl/compat/opensolaris/sys/vnode.h

```diff
@@ -268,7 +268,8 @@ vn_rename(char *from, char *to, enum uio_seg seg)
 
 	ASSERT(seg == UIO_SYSSPACE);
 
-	return (kern_renameat(curthread, AT_FDCWD, from, AT_FDCWD, to, seg));
+    return (kern_renameat(curthread, AT_FDCWD, (__cheri_tocap const char * __capability) from, AT_FDCWD,
```

Comment:
Weird indentation.

Reply:
Anyway, this seems to be used only by zfs. Was this needed to silence compiler errors because the header is somehow included by dtrace?

Reply:
Yeah I just did that following a suggestion from the compiler, without thinking much. 
It is also used by zfs, but from what I've seen the callers are not passing a capability:
sys/cddl/contrib/opensolaris/uts/common/fs/zfs/spa_config.c:202
sys/cddl/contrib/opensolaris/uts/common/fs/zfs/zfs_vnops.c:3892

..how should I do?

Reply:
in `spa_config.c`, `zfs_context.h` is included, and there is `#define	vn_rename(from, to, seg)	rename((from), (to))` there. So probably only the other one is relevant